### PR TITLE
Initial commit of the tagging-migration-verifier

### PR DESCRIPTION
**Pull request by tijmenb**
Context: [we're migrating applications so that tagging is done via the publishing-api](https://trello.com/c/A4P3V9gK/414-epic-migrating-apps-to-new-tagging-infrastructure). This will enable us to build really cool tagging tools and serve everything from the content-store.

While we want to migrate the apps that send taggings, we don't want to migrate the apps that consume the tags (Whitehall, Rummager, and maybe more) yet.

To achieve that, we copy the taggings sent to the publishing-api to panopticon/contentapi, so that they are available as usual (https://github.com/alphagov/panopticon/pull/307 and https://github.com/alphagov/panopticon/pull/313).

The current complexity in data flow means that we have to be careful.

This **tagging-migration-verifier** is a script that verifies that the taggings in the content-store
(https://github.com/alphagov/content-store/pull/170) are in sync with the contentapi (https://github.com/alphagov/govuk_content_api/pull/233).

```
$ bin/verify_migrated_apps
```

This command will be run periodically by a Jenkins script. When it finds a difference between the taggings in content-store and content-api, it will exit with a exit code `1`. This will trigger an alert to our
monitoring stack (Icinga), so that we can investigate and fix any mismatches.

Manual testing on preview can be done with:

```
$ git clone https://github.com/alphagov/tagging-migration-verifier.git
$ cd tagging-migration-verifier
$ # git checkout initial-implementation
$ APP_DOMAIN=preview.alphagov.co.uk bin/verify_migrated_apps
```

Trello: https://trello.com/c/DLakGcP0
Puppet PR: https://github.gds/gds/puppet/pull/3705

### CODE REVIEW

File: lib/compare.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+class Compare
```
Comment (jackscotti) [inline lib/compare.rb]: Why is this a `Compare` rather than a `Comparer`?

Comment (tijmenb) [inline lib/compare.rb]: Maybe it's more a `Diff`.

File: lib/comparer.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  def only_b
+    b - a
+  end
+
+  def prepare(hash)
+    hash.each do |content_id, links_hash|
+      links_hash.each do |tag_name, link_ids|
+        links_hash[tag_name] = link_ids.sort
+      end
+    end.to_a
```
Comment (ghost) [inline lib/comparer.rb]: Style guidelines call { block }.to_a rather than do-end

Comment (tijmenb) [inline lib/comparer.rb]: Updated.

**Pull request merged**